Sources/NIO/Embedded.swift

```diff
@@ -359,7 +354,7 @@ public class EmbeddedChannel: Channel {
         }
     }
 
-    private func readFromBuffer(buffer: inout [IOData]) -> IOData? {
+    private func readFromBuffer(buffer: inout [NIOAny]) -> NIOAny? {
```

Comment:
If you make the change suggested above, this function can just go away and we can use the generic variant instead.